src/overlays/actors/ovl_Obj_Jgame_Light/z_obj_jgame_light.c

```diff
@@ -28,29 +39,172 @@ const ActorInit Obj_Jgame_Light_InitVars = {
+
+void ObjJgameLight_Init(Actor* thisx, PlayState* play) {
+    ObjJgameLight* this = THIS;
+    LightInfo* lights = &this->lightInfo;
+
+    Actor_SetScale(&this->actor, 1.0f);
+    func_800B4AEC(play, &this->actor, 50.0f);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B4B50, 1.0f);
+    Collider_InitCylinder(play, &this->collider);
+    Collider_SetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+    Lights_PointGlowSetInfo(lights, this->actor.world.pos.x, (this->actor.world.pos.y + 70.0f), this->actor.world.pos.z,
+                            255, 255, 180, -1);
+    this->lightNode = LightContext_InsertLight(play, &play->lightCtx, &this->lightInfo);
+    Actor_SetFocus(&this->actor, 60.0f);
+    this->actor.colChkInfo.health = 0;
+    this->unk_1B8 = 0;
+    this->unk_1B6 = 0;
+    this->lightRadius = 0;
+    this->alpha = 0;
+    this->signal = OBJJGAMELIGHT_NONE;
+    this->unk_1A8 = 0.0f;
+}
+
+void ObjJgameLight_Destroy(Actor* thisx, PlayState* play) {
+    ObjJgameLight* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+    LightContext_RemoveLight(play, &play->lightCtx, this->lightNode);
+}
+
+void func_80C15474(ObjJgameLight* this, PlayState* play) {
+    u8 temp_a1;
+
+    if (this->actor.colChkInfo.health & OBJLUPYGAMELIFT_IGNITEFIRE && (this->unk_1B6 == 0)) {
+        if (this->lightRadius < 0xA0) {
+            this->lightRadius += 0x28;
+        } else {
+            this->lightRadius = 0xC8;
+            this->unk_1B6 = 1;
+        }
+        if (this->unk_1A8 < 0.7f) {
+            this->unk_1A8 += 0.3f;
+        } else {
+            this->unk_1A8 = 1.0f;
+        }
+    } else if (this->actor.colChkInfo.health & OBJLUPYGAMELIFT_SNUFF_FIRE) {
+        if (this->lightRadius > 0x28) {
+            this->lightRadius -= 0x28;
+        } else {
+            this->lightRadius = -1;
+            if (this->unk_1A8 == 0.0f) {
+                this->unk_1B6 = 0;
+                this->actor.colChkInfo.health &= ~OBJLUPYGAMELIFT_IGNITEFIRE;
+                this->actor.colChkInfo.health &= ~OBJLUPYGAMELIFT_SNUFF_FIRE;
+            }
+        }
+        if (this->unk_1A8 > 0.3f) {
+            this->unk_1A8 -= 0.3f;
+        } else {
+            this->unk_1A8 = 0.0f;
+        }
+    }
+    if (this->unk_1A8 > 0.1f) {
+        func_800B9010(&this->actor, NA_SE_EV_TORCH - SFX_FLAG);
+    }
+    temp_a1 = (s32)(Rand_ZeroOne() * 127.0f) + 128;
+    Lights_PointSetColorAndRadius(&this->lightInfo, temp_a1, temp_a1 * 0.7f, 0, this->lightRadius);
+}
+
+void ObjJgameLight_UpdateCollision(ObjJgameLight* this, PlayState* play) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    CollisionCheck_SetAC(play, &play->colChkCtx, &this->collider.base);
+}
+
+void func_80C15718(ObjJgameLight* this, PlayState* play) {
+    if (this->actor.colChkInfo.health & OBJLUPYGAMELIFT_IGNITEFIRE && !(this->unk_1B8 & 1)) {
```

Comment:
```suggestion
    if ((this->actor.colChkInfo.health & OBJLUPYGAMELIFT_IGNITEFIRE) && !(this->unk_1B8 & OBJLUPYGAMELIFT_IGNITEFIRE)) {
```
unk_1B8 looks to be `prevHealth`, so can use the same define.